common/f3d_materials.md

```diff
@@ -0,0 +1,193 @@
+
+## RGB/Alpha Dither
+
+Dithering affects the final color that is output to the framebuffer. The affect is used to reduce artifacts and banding when converting higher precision source colors from varoius sources into the lower precision framebuffer. 8 bits are used per RGB component in the RDP pipeline, and it is written to a 5 bit per channel framebuffer. Try changing the types of dithering if banding occurs.
+
+* Pattern - Alpha only option. Follows the same dithering as rgb or uses Bayer
+* NotPattern - Alpha only option. Dithers with the opposite pattern as "Pattern"
+* Magic Square - RGB only option. Uses a magic square for dithering
+* Bayer - RGB only option. Uses a Bayer matrix for dithering
+* Noise - Random dithering
+* Enable - Option just for backwards compatibility. Equivalent to "Noise"
+* Disable - No dithering is uses.
+
+## Chroma Key
+
+Chroma keying allows you to specify colors as pixel alpha. A combiner with equation {eq}`chroma_key_combiner_f3d_mat` is recommended for use and should be used in the last available cycle. The combiner will use the `A` value put in variable A as the color output, and the alpha out will be the min value from each color channel of {eq}`key_alpha_calc_f3d_mat`.
+In order to use chroma key, `Key` must be selected in the chroma key dropdown and [alpha_cvg_sel]{.key_text} must be off in the chosen render mode.
+
+:::{card}
+$$
+combined\_color = (A - key\_center)*key\_scale + 0
+$$(chroma_key_combiner_f3d_mat)
+$$
+key\_alpha = clamp(0, key\_width - abs(combined\_color), 255)
+$$(key_alpha_calc_f3d_mat)
+:::
+
+## Texture Convert/Filter
+
+Texture Convert allows for the enabling of YUV conversion, and texture filtering. Filtering type is controlled by the Texture Filter option. For filtering, billinear is standard, and point is for no filtering. Average is a special type of filter only for pixel/texel aligned rendering.
+
+## Texture LOD/Detail
+
+Texture LOD enables the use of Level of Detail (LOD) tiles. Texture Detail controls the type of LOD used. LOD calculations start with texture 0 being tile 0, and increase based on the texel/pixel ratio up to a max based on the number of mipmaps setting. The Min LOD Ratio is used when in LOD mode clamp, and governs how low the LOD can get while under magnification (texel/pixel < 1). For other LOD modes, this value is locked at 0.5.
+
+## Texture Perspective Correction
+
+This enables the correction of rasterized triangles with skew. Without correction, triangle colors would distort when triangles are moved in relation to the camera.
+
+## Cycle Type
+
+1 and 2 Cycle enable the full rendering pipeline and are the default options, with 2 cycle enabling additional options for combiner and render mode selection. 2 Cycle is required for loading multiple textures or doing certain types of blending (like with fog).
```

Comment:
> 2 Cycle is required for loading multiple textures

is it? :thinking:

Reply:
The manual always says to use 2cycle when multitexturing but they don't explicitly state you need two cycle unless you are using LoD.
That said I did test this and I got different results in one and two cycle mode using the same combiner.
![image](https://github.com/Fast-64/fast64-docs/assets/6765126/ea2dc061-29bc-46d3-995a-3b83a224d2bd)
All of the tests are opaque, texel0*texel1, top is 2 cycle, with the left doing the combining in the first cycle, and right in the second.
The bottom ones just swap the order of textures.
So on two cycle we get the expected result, and in one cycle there is something wonky going on. I did an additional test got this:
![image](https://github.com/Fast-64/fast64-docs/assets/6765126/2f9071f7-64b5-46c4-8f5b-7b8a2055e8cd)
Basically the right is texel1*texel1, the left texel0*texel0. The top loads grass then sand, while the bottom loads sand then grass. The expected output if everything loaded properly is shown in the fast64 preview.
Basically it seems like texel1 has the same value as texel0, as all tiles only show the output of the first load * first load.
I am using angrylion to test this out.
This matches my expectations a little but I'm not sure if I have anything concrete saying that texel1 needs 2 cycle.